rust/utils/src/compression.rs

```diff
@@ -0,0 +1,33 @@
+use std::str::FromStr;
+use anyhow::anyhow;
+
+pub const CAS_CONTENT_ENCODING_HEADER: &str = "xet-cas-content-encoding";
```

Comment:
Where to use this value?

Reply:
This will be used in 2 spots:
- on puts with compression from the client
- on get responses from the server. client will use this header to determine the encoding.

We will not be using the standard HTTP Content-Encoding header due to the fact that LZ4 is not one of the "official" supported encodings.